fix authentication field and password field validators to work

kasongoyo opened the pull request:
This fix involve replacing mongoose-valid8 with [https://github.com/chriso/validator.js](url)  as the result the Schema type email is no longer required

kasongoyo committed:
fix authentication field and password field validators to work

kasongoyo committed:
Fix bugs and revamp features

	- Fix the issue of all default options to disappear when you overide some.
	- Add ability to disable account lock feature.
	- Improve readMe to cover for plugin options.
	- Fix bug that instance authenticate method never check if account is locked.

lykmapipo commented:
@kasongoyo may you pleas add failed spec so that we can knwo waht was failing before merge

lykmapipo commented on lib/morphs/authenticable.js:
```diff
@@ -46,26 +44,34 @@ module.exports = exports = function Authenticable(schema, options) {
+            lowercase: true,
+            //trim authentication field
+            trim: true,
+            validate: [validator.isEmail, 'Invalid email address'],
```
This field can not be email address as it provided by implementer of irina. Consider removing it

kasongoyo committed:
fix authenticate instance method to detect if instance is not comfirmable

kasongoyo committed:
Add support for Promise based functions

	- Promisify registerable
	- promisify confirmable
	- Promisify lockable
	- Promisify authenticable

kasongoyo committed:
Promisification of recoverable and trackable

kasongoyo committed:
Merge branch 'feature/registerable' into develop

kasongoyo committed:
update package version
